assets/src/components/mapPage/routePropertiesCard.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  RoutePattern,
+  RoutePatternId,
+  DirectionName,
+  DirectionId,
+  RouteId,
+  Route,
+  ByRoutePatternId,
+} from "../../schedule"
+import { Card } from "../card"
+import { RoutePill } from "../routePill"
+
+const sortRoutePatterns = (routePatterns: RoutePattern[]): RoutePattern[] =>
+  routePatterns.sort((rp1, rp2) => rp1.sortOrder - rp2.sortOrder)
+
+const DirectionPicker = ({
+  selectedRoutePattern,
+  routePatterns,
+  selectRoutePattern,
+  directionNames,
+}: {
+  selectedRoutePattern: RoutePattern
+  routePatterns: ByRoutePatternId<RoutePattern>
+  selectRoutePattern: (routePattern: RoutePattern) => void
+  directionNames: {
+    0: DirectionName
+    1: DirectionName
+  }
+}) => {
+  return (
+    <div className="m-route-properties-card__direction-picker">
+      {
+        // TODO: Should we always show both buttons? Some routes only go one direction?
+      }
+      {[0, 1].map((directionId) => (
+        <div key={directionId}>
+          <input
+            type="radio"
+            id={`direction-radio-${directionId}`}
+            name="direction"
+            defaultChecked={selectedRoutePattern.directionId === directionId}
+            onChange={() => {
+              const targetDirectionId = directionId === 0 ? 1 : 0
+              const targetRoutePattern =
+                sortRoutePatterns(Object.values(routePatterns)).find(
+                  (routePattern) =>
+                    routePattern.directionId === targetDirectionId
+                ) || null
+
+              if (targetRoutePattern) {
+                selectRoutePattern(targetRoutePattern)
+              }
+            }}
+          />
+          <label htmlFor={`direction-radio-${directionId}`}>
+            {directionNames[directionId as DirectionId]}
+          </label>
+        </div>
+      ))}
+    </div>
+  )
+}
+
+const VariantPicker = ({
+  routePatterns,
+  selectedRoutePatternId,
+  selectRoutePattern,
+}: {
+  routePatterns: RoutePattern[]
+  selectedRoutePatternId: RoutePatternId
+  selectRoutePattern: (routePattern: RoutePattern) => void
+}): JSX.Element => {
+  {
+    return (
+      <>
+        {sortRoutePatterns(routePatterns).map((routePattern) => (
+          <div
+            className="m-route-properties-card__variant-picker"
+            key={routePattern.id}
+          >
+            <input
+              type="radio"
+              id={`variant-radio-${routePattern.id}`}
+              name="variant"
+              defaultChecked={routePattern.id === selectedRoutePatternId}
+              onChange={() => selectRoutePattern(routePattern)}
+            />
+
+            <label htmlFor={`variant-radio-${routePattern.id}`}>
+              <div>
+                {routePattern.name}
+                {routePattern.timeDescription && routePattern.timeDescription}
+              </div>
+            </label>
+          </div>
+        ))}
+      </>
+    )
+  }
+}
+
+const RoutePropertiesCard = ({
+  routeId,
```

Comment:
**suggestion:** if you want to minimize the number of props, you could get `routeId` directly from `selectedRoutePattern` once that's known.